src/ode_solver/explicit_ode_solver.cpp

```diff
@@ -107,10 +60,39 @@ void ExplicitODESolver<dim,real,MeshType>::allocate_ode_system ()
     this->solution_update.reinit(this->dg->right_hand_side);
     this->dg->evaluate_mass_matrices(do_inverse_mass_matrix);
 
-    this->rk_stage.resize(4);
-    for (int i=0; i<4; i++) {
+    Parameters::ODESolverParam ode_param = ODESolverBase<dim,real,MeshType>::all_parameters->ode_solver_param;
+    const int rk_order = ode_param.runge_kutta_order;
+    this->rk_stage.resize(rk_order);
+    for (int i=0; i<rk_order; i++) {
         this->rk_stage[i].reinit(this->dg->solution);
     }
+
+    //Assigning butcher tableau
+    this->butcher_a.reinit(rk_order,rk_order);
+    this->butcher_b.reinit(rk_order);
+    if (rk_order == 3){
+        //RKSSP3
+        const double butcher_a_values[9] = {0,0,0,1.0,0,0,0.25, 0.25, 0};
```

Comment:
`butcher_a_values` seems to always require the first row to be zeros, if that is always the case in the future, is there a way you could modify the implementation to avoid having to specify these? for example, this line would simply be `const double butcher_a_values[9] = {1.0,0,0,0.25, 0.25, 0};`

Reply:
and if you use the coefficient matrix idea I mentioned earlier, it would be `const double runge_kutta_coefficients[rk_order*rk_order] = {1.0,0,0, 0.25,0.25,0, 1.0/6.0, 1.0/6.0, 2.0/3.0};` then these would fill the `runge_kutta_coefficient_matrix`